components/net/resource_thread.rs

```diff
@@ -118,14 +118,20 @@ fn create_http_states(config_dir: Option<&Path>) -> (Arc<HttpState>, Arc<HttpSta
         read_json_from_file(&mut cookie_jar, config_dir, "cookie_jar.json");
     }
 
-    let ca_file = match opts::get().certificate_path {
-        Some(ref path) => PathBuf::from(path),
-        None => resources_dir_path()
-            .expect("Need certificate file to make network requests")
-            .join("certs"),
+    let certs = match opts::get().certificate_path {
+        Some(ref path) => {
+            let mut txt = String::new();
+            let mut file = File::open(PathBuf::from(path))
+                .expect("Couldn't not find certificate file");
+            file.read_to_string(&mut txt).expect("Cant read certificate");
```

Comment:
This can use [`fs::read_to_string`](https://doc.rust-lang.org/nightly/std/fs/fn.read_to_string.html).